src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -404,22 +405,38 @@ internal void VerifyMutable()
 
         private protected volatile bool _isConfigured;
         private readonly object _configureLock = new object();
+        private ExceptionDispatchInfo? _cachedConfigureError;
 
         internal bool IsConfigured => _isConfigured;
 
         internal void EnsureConfigured()
         {
-            if (_isConfigured)
-                return;
+            if (!_isConfigured)
+                ConfigureLocked();
 
-            lock (_configureLock)
+            void ConfigureLocked()
             {
-                if (_isConfigured)
-                    return;
+                _cachedConfigureError?.Throw();
+
+                lock (_configureLock)
+                {
+                    if (_isConfigured)
+                        return;
+
+                    _cachedConfigureError?.Throw();
 
-                Configure();
+                    try
+                    {
+                        Configure();
```

Comment:
Does try..catch have any perf impact? It's not our main concern but it will increase static initialization time if it does. Should we just let it through instead? (no change) I think it's ok if few variables get re-initialized in the exception state

Reply:
Theoretically it does, but this is already in the critical section of a lock, so this will not have any impact on performance.